test/integration/PoolSwap.t.sol

```diff
@@ -1,92 +1,204 @@
-import {Test} from 'forge-std/Test.sol';
+import {GPv2TradeEncoder} from './GPv2TradeEncoder.sol';
+import {IERC20} from '@cowprotocol/interfaces/IERC20.sol';
+import {GPv2Interaction} from '@cowprotocol/libraries/GPv2Interaction.sol';
+import {GPv2Order} from '@cowprotocol/libraries/GPv2Order.sol';
+import {GPv2Trade} from '@cowprotocol/libraries/GPv2Trade.sol';
+import {GPv2Signing} from '@cowprotocol/mixins/GPv2Signing.sol';
+import {BCoWConst} from 'contracts/BCoWConst.sol';
+import {BCoWPool} from 'contracts/BCoWPool.sol';
+import {GasSnapshot} from 'forge-gas-snapshot/GasSnapshot.sol';
+import {Test, Vm} from 'forge-std/Test.sol';
+import {IBCoWPool} from 'interfaces/IBCoWPool.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
 
-import {BFactory} from 'contracts/BFactory.sol';
-import {IBPool} from 'interfaces/IBPool.sol';
+abstract contract PoolSwapIntegrationTest is Test, BCoWConst, GasSnapshot {
+  IBCoWPool public pool;
 
-import {GasSnapshot} from 'forge-gas-snapshot/GasSnapshot.sol';
+  IERC20 public dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+  IERC20 public weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
+  address public solver = address(0xa5559C2E1302c5Ce82582A6b1E4Aec562C2FbCf4);
 
-abstract contract PoolSwapIntegrationTest is Test, GasSnapshot {
-  BFactory public factory;
-  IBPool public pool;
+  address public controller = makeAddr('controller');
+  address public lp = makeAddr('lp');
+  Vm.Wallet swapper = vm.createWallet('swapper');
 
-  IERC20 public tokenA;
-  IERC20 public tokenB;
+  bytes32 public constant APP_DATA = bytes32('exampleIntegrationAppData');
 
-  address public lp = address(420);
-  address public swapper = address(69);
+  uint256 public constant HUNDRED_UNITS = 100 ether;
+  uint256 public constant ONE_UNIT = 1 ether;
+  uint256 public constant WETH_AMOUNT = 0.096397921069149814e18;
+  uint256 public constant DAI_AMOUNT = 0.5e18;
 
   function setUp() public {
-    tokenA = IERC20(address(deployMockERC20('TokenA', 'TKA', 18)));
-    tokenB = IERC20(address(deployMockERC20('TokenB', 'TKB', 18)));
+    vm.createSelectFork('mainnet', 20_012_063);
 
-    deal(address(tokenA), address(lp), 100e18);
-    deal(address(tokenB), address(lp), 100e18);
+    deal(address(dai), address(lp), HUNDRED_UNITS);
+    deal(address(weth), address(lp), HUNDRED_UNITS);
 
-    deal(address(tokenA), address(swapper), 1e18);
-
-    factory = new BFactory();
+    deal(address(dai), address(swapper.addr), ONE_UNIT);
 
     vm.startPrank(lp);
-    pool = factory.newBPool();
-
-    tokenA.approve(address(pool), type(uint256).max);
-    tokenB.approve(address(pool), type(uint256).max);
-
-    pool.bind(address(tokenA), 1e18, 2e18); // 20% weight?
-    pool.bind(address(tokenB), 1e18, 8e18); // 80%
-
+    // TODO: deploy using BCoWFactory
+    pool = new BCoWPool(address(settlement));
```

Comment:
in this file we should be testing `BPool`
in the other file `BCowPool.t.sol` we should override `BPool` delpoyment for a `BCoWPool`